modules/claims_api/app/controllers/claims_api/v0/forms/power_of_attorney_controller.rb

```diff
@@ -17,16 +17,17 @@ class PowerOfAttorneyController < ClaimsApi::BaseFormController
         before_action :validate_documents_page_size, only: %i[upload]
 
         def submit_form_2122
-          power_of_attorney = ClaimsApi::PowerOfAttorney.create(
-            status: ClaimsApi::PowerOfAttorney::PENDING,
-            auth_headers: auth_headers,
-            form_data: form_attributes,
-            source_data: {
-              name: request.headers['X-Consumer-Username'],
-              icn: Settings.bgs.external_uid,
-              email: Settings.bgs.external_key
-            }
-          )
+          power_of_attorney = ClaimsApi::PowerOfAttorney.find_by(header_md5: header_md5)
+          unless %w[submitted pending].include? power_of_attorney.try(:status)
+            power_of_attorney = ClaimsApi::PowerOfAttorney.create(
+              status: ClaimsApi::PowerOfAttorney::PENDING,
+              auth_headers: auth_headers,
+              form_data: form_attributes,
+              source_data: source_data,
+              header_md5: header_md5
+            )
+          end
+
           power_of_attorney = ClaimsApi::PowerOfAttorney.find_by(md5: power_of_attorney.md5) unless power_of_attorney.id
           power_of_attorney.save!
```

Comment:
This is a bit hard to read for me, so I started rewriting it to help me read it (don't take offense, this is just how i read stuff sometimes):
```
def submit_form_2122
  @power_of_attorney = ClaimsApi::PowerOfAttorney.find_by(header_md5: header_md5)

  unless @power_of_attorney && poa_is_submitted_or_pending
    @power_of_attorney = ClaimsApi::PowerOfAttorney.create(
      status: ClaimsApi::PowerOfAttorney::PENDING,
      auth_headers: auth_headers,
      form_data: form_attributes,
      source_data: source_data,
      header_md5: header_md5
    )
    @power_of_attorney = ClaimsApi::PowerOfAttorney.find_by(md5: @power_of_attorney.md5)
  end

  @power_of_attorney.save!

  ClaimsApi::PoaUpdater.perform_async(@power_of_attorney.id)

  render json: @power_of_attorney, serializer: ClaimsApi::PowerOfAttorneySerializer
end


def poa_is_submitted_or_pending
  @power_of_attorney.status.in? %w[submitted pending]
end
```
in doing so, the _find by md5 _ line seems weird.
should it be
```
def submit_form_2122
  @power_of_attorney = ClaimsApi::PowerOfAttorney.find_by(header_md5: header_md5)

  unless @power_of_attorney && poa_is_submitted_or_pending
    @power_of_attorney = ClaimsApi::PowerOfAttorney.create(
      status: ClaimsApi::PowerOfAttorney::PENDING,
      auth_headers: auth_headers,
      form_data: form_attributes,
      source_data: source_data,
      header_md5: header_md5
    )
    @power_of_attorney = ClaimsApi::PowerOfAttorney.find_by(md5: @power_of_attorney.md5) || @power_of_attorney #HERE
  end

  @power_of_attorney.save!

  ClaimsApi::PoaUpdater.perform_async(@power_of_attorney.id)

  render json: @power_of_attorney, serializer: ClaimsApi::PowerOfAttorneySerializer
end


def poa_is_submitted_or_pending
  @power_of_attorney.status.in? %w[submitted pending]
end
```
or, in your original:
```
```
def submit_form_2122
  power_of_attorney = ClaimsApi::PowerOfAttorney.find_by(header_md5: header_md5)
  unless %w[submitted pending].include? power_of_attorney.try(:status)
    power_of_attorney = ClaimsApi::PowerOfAttorney.create(
      status: ClaimsApi::PowerOfAttorney::PENDING,
      auth_headers: auth_headers,
      form_data: form_attributes,
      source_data: source_data,
      header_md5: header_md5
    )
  end

  unless power_of_attorney.id # HERE
    power_of_attorney = ClaimsApi::PowerOfAttorney.find_by(md5: power_of_attorney.md5) || power_of_attorney
  end
  power_of_attorney.save!

  ClaimsApi::PoaUpdater.perform_async(power_of_attorney.id)

  render json: power_of_attorney, serializer: ClaimsApi::PowerOfAttorneySerializer
end
```

Reply:
@edmangimelli I like the direction you are suggesting especially since after we wrap this up it likely wont be touching for a while. When we get back into it, the more readable the better

Reply:
I personally prefer not to use instance variables in controllers unless it's absolutely necessary, would a good middle ground here be to do

unless power_of_attorney&.status.in? %w[submitted pending]

which allows you to both check the existence of power of attorney without needing to repeat?

I can also bring the md5 find by into the loop for sure.

Reply:
my big concern is that is seems like this line:
```
power_of_attorney = ClaimsApi::PowerOfAttorney.find_by(md5: power_of_attorney.md5) unless power_of_attorney.id
```
obliterates the newly created poa (*if* one was created)
so I'm thinking it should be 
```
poa = find_by_md5 || poa unless poa.id
```
(except with all the correct names)

Reply:
yeah, i'm not familiar with this repos (or teams) conventions yet (re: instance variables).
I've spent nearly all my time in the `caseflow` repo up until now

Reply:
The .id check is to see if the object is persisted or not so still a good thing to check against it instead of whether or not the object just exists.

Reply:
oh wait.. I was thinking `new` was being used (not `create` --i thought for a sec that we had a poa object that hadn't been commited to the DB).
nm.
hmmm. that line still seems weird though. won't `power_of_attorney` always have an id? it's either found (first line of method) or created (in the first unless).

Reply:
@bastosmichael just for readability, can you change the .id check to .persisted? https://apidock.com/rails/ActiveRecord/Persistence/persisted

Reply:
Changed to persisted and moved things around as requested.